distribution/lib/Standard/Base/0.0.0-dev/src/Network/Reload_Detector.enso

```diff
@@ -0,0 +1,32 @@
+import Standard.Base.Data.Boolean.Boolean
+import Standard.Base.Nothing.Nothing
+import Standard.Base.Runtime.Managed_Resource.Managed_Resource
+import Standard.Base.Runtime.Ref.Ref
+
+## PRIVATE
+   This is used by ReloadDetector.java to create a `Managed_Resource` that is
+   finalized when the reload button is pressed.
+
+   The `on_finalize` function and the `clear` method both write `Nothing` to the
+   ref. This is a signal that a reload has happenend. `on_finalize` is called by
+   the engine when a reload happens. `clear` is only for testing, to simulate a
+   reload.
+
+   The `0` value stored in the ref is not used; it just has to be something
+   other than Nothing.
+type Reload_Detector
+    private Value mr:Managed_Resource
+
+    new -> Reload_Detector =
+        ref = Ref.new 0
+        on_finalize ref = ref.put Nothing
+        mr = Managed_Resource.register ref on_finalize Boolean.True
+        Reload_Detector.Value mr
+
+    get self = self.mr.with .get
+
+    clear self =
+        self.mr.with (ref-> ref.put Nothing)
+        Nothing
```

Comment:
So that's why I was asking to have a Panic and not a dataflow error. If by any chance (through correct or incorrect code) the `mr.with` block raises an error, this is a place where we will discard and lose it.
```suggestion
        self.mr.with (ref-> ref.put Nothing) . if_not_error Nothing
```

This would have been much simpler if this was a Panic in the first place...

(cc: @JaroslavTulach)

Reply:
In this case we however have a valid code. `clear` of already GCed reference is a no-op. Which is exactly this:
- ignore potential error
- return `Nothing`

sequence does. Maybe it was @GregoryTravis's intention?

Reply:
What can we learn from this? Let's **claim**: Use of `Error` _encourages robust code_, as it is idempotent to double cleanup like the one performed here.

Reply:
It was a mistake -- I didn't realize the resource was GC'd as well as having the finalizer run. I don't think it's enough to check for Nothing -- what if something changes, and the Nothing is stored before the release of the reference, then it will no longer be nothing.

Reply:
I have updated it to use `catch` to detect the finalization / collection.